### ISSUE TITLE
Minecart train sometimes reverses direction

### ISSUE THREAD

**Issue by lgnmcrules** [Kipama/Modern_Minecarts#29]
![image](https://github.com/Kipama/Modern_Minecarts/assets/47130259/0365b867-b241-4bbf-bbd5-e511bc9e9808)

Occasionally, the furnace minecart train will randomly reverse direction.  It was not related to colliding with the cat in the screenshot.  Trains also get various issues if one of the carts was spawned on a sloped rail, but not all of them.

**Comment by Kipama** [comment]
Did this happen around 180° Corners?

**Comment by lgnmcrules** [comment]
Yes, around 180 corners with nothing there this happens occasionally.  This design here was my attempt at making a quick turn for trains.

**Comment by Kipama** [comment]
This is an unfortunate side effect of the way trains work. 180° turns with less than a 2 block gap are unstable and will likely remain so, as a fix would require a full rewrite of the train logic.
